joomla/joomla-cms
# Mod languages performance improvements

#### Description

Mod_languages (the module with the language flags), is one of the modules that can be in all pages that has slower performances (if 3 or more languages > 10ms).

So, this PR makes some code changes to get some small performance improvements.

No behaviour is changed.
##### Performance Tests

| Published Languages | Before PR | After PR |
| :-- | :-: | :-: |
| 1 | ~5.5 ms | ~3.7 ms |
| 2 | ~7.5 ms | ~4.0 ms |
| 3 | ~10 ms | ~4.2 ms |
| 4 | ~12 ms | ~4.4 ms |
| 5 | ~15 ms | ~5.3 ms |

Note: Tests were made with Joomla Debug Console in PHP 5.6 - no caching
#### How to test
1. Fresh Joomla install with multilanguage (install 5 or more languages)
2. Apply this patch.
3. Verify all the flags in mod_languages are working fine

To check the performance improvements, turn on Joomla debugging and check the `Application: afterRenderModule mod_languages (xxxxxx)` execution time, before and after the PR
#### Observations

Suggestions, code reviews and improvements are welcome.
